code/api/src/modules/product/mutations.test.js

```diff
@@ -0,0 +1,127 @@
+import request from 'supertest'
+import express from 'express'
+import schema from '../../setup/schema'
+import graphqlHTTP from 'express-graphql'
+import authentication from '../../setup/authentication'
+import models from '../../setup/models'
+import { isType } from 'graphql'
+import bcrypt from 'bcrypt'
+import config from '../../config/server.json'
+import db from '../../setup/database'
+
+
+
+
+describe('product mutations', () => {
+
+  let server;
+  let token;
+  let productId;
+
+  beforeAll(async () => { // get the server running and attached for each of our future tests
+    server = express();
+
+    server.use(authentication);
+
+    server.use(
+      '/',
+      graphqlHTTP(request => ({
+        schema: schema, // import the schema
+        graphiql: true,
+        context: {
+          auth: {
+            user: request.user,
+            isAuthenticated: request.user && request.user.id > 0
+          }
+        }
+      })
+      )
+    )
+    await models.Product.destroy({ where: {} })
+  })
+
+  beforeEach(async () => {
+    const adminUser = {
+      id: 5,
+      name: "adminProduct",
+      email: "admin@crate.com",
+      password: bcrypt.hashSync('123456', config.saltRounds),
+      role: "ADMIN",
+      createdAt: new Date(),
+      updatedAt: new Date(),
+      style: "Casual"
+    };
+
+    await models.User.destroy({where: {}});
+    await models.User.create(adminUser);
```

Comment:
this is a bit of a new standard for the test suit. destroy everything, then re-create it